internal/config/validation.go

```diff
@@ -17,12 +18,42 @@ import (
 	"github.com/edgelesssys/constellation/v2/internal/versions"
 	ut "github.com/go-playground/universal-translator"
 	"github.com/go-playground/validator/v10"
+	"golang.org/x/mod/semver"
 )
 
 func validateK8sVersion(fl validator.FieldLevel) bool {
 	return versions.IsSupportedK8sVersion(fl.Field().String())
 }
 
+func registerInvalidK8sVersionError(ut ut.Translator) error {
+	return ut.Add("invalid_k8s_version", "{0} specifies an unsupported Kubernetes version. {1}", true)
+}
+
+func translateInvalidK8sVersionError(ut ut.Translator, fe validator.FieldError) string {
+	validVersions := make([]string, len(versions.VersionConfigs))
+	i := 0
+	for k := range versions.VersionConfigs {
+		validVersions[i] = string(k)
+		i++
+	}
+	validVersionsSorted := semver.ByVersion(validVersions)
+	sort.Sort(validVersionsSorted)
+
+	var errorMsg string
+	configured := fe.Value().(string)
+	maxVersion := validVersionsSorted[len(validVersionsSorted)-1]
+	if configured < validVersionsSorted[0] {
+		errorMsg = fmt.Sprintf("You can upgrade to these versions with your current Constellation CLI: %s.", validVersionsSorted)
```

Comment:
```suggestion
		errorMsg = fmt.Sprintf("Upgrade to one of these versions with this CLI: %s.", validVersionsSorted)
```
I'd omit "Constellation" since it is probably clear that it is the Constellation CLI and to align with the message below. Also shortened it a bit and increased the nudging to emphasize the needed action to resolve the error.

Reply:
I think we should leave out the "You can upgrade" part, since this error will occur even when you try to create a new Constellation.
Maybe just align it with the version-too-new error, and tell the user that the configured version is not supported?

Reply:
Tbh I wouldn't want to leave out the upgrade part because it is supposed to support users when they upgraded their CLI while upgrading a cluster. 
Imagine your first CLI tells you to upgrade the CLI. Now the second CLI tells you sth like "please select a supported version". I would feel like there is a gap between both CLIs in that case. (?)

Reply:
That would be a user error, wouldn't it?
A user is expected to updated their config when they want to upgrade the cluster.
Failure to do so can result in invalid configurations (hence the "This version is not supported").

I also feel like reporting a "You can upgrade to these version with your CLI" when a user tries to create a completely new cluster may lead to some questions that would be avoided by simply giving a more general response.

Reply:
After another live discussion we decided to extend the versionapi so it can server information on cli-k8s compatibility. This removes the original motivation for the initial error message. Changed both error messages to generic messages.